hal/src/trackerm/network/network.cpp

```diff
@@ -0,0 +1,260 @@
+        ncpClient_ = std::move(ncpClient);
+        return 0;
+    }
+};
+
+bool netifCanForwardIpv4(netif* iface) {
+    if (iface && netif_is_up(iface) && netif_is_link_up(iface)) {
+        auto addr = netif_ip_addr4(iface);
+        auto mask = netif_ip_netmask4(iface);
+        auto gw = netif_ip_gw4(iface);
+        if (!ip_addr_isany(addr) && !ip_addr_isany(mask) && !ip_addr_isany(gw)) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+
+class CellularNetworkManagerInit {
+public:
+    CellularNetworkManagerInit() {
+        const int r = init();
+        SPARK_ASSERT(r == 0);
+    }
+
+    CellularNetworkManager* instance() const {
+        return mgr_.get();
+    }
+
+private:
+    std::unique_ptr<CellularNcpClient> client_;
+    std::unique_ptr<CellularNetworkManager> mgr_;
+
+    int init() {
+        // Get active SIM card
+        SimType sim = SimType::INVALID;
+        CHECK(CellularNetworkManager::getActiveSim(&sim));
+        CellularNcpClientConfig conf;
+        conf.simType(sim);
+        conf.ncpIdentifier(platform_primary_ncp_identifier());
+        conf.eventHandler(PppNcpNetif::ncpEventHandlerCb, pp3);
+        conf.dataHandler(PppNcpNetif::ncpDataHandlerCb, pp3);
+        // Initialize NCP client
+        std::unique_ptr<CellularNcpClient> client;
+        client.reset(new(std::nothrow) QuectelNcpClient);
+        CHECK_TRUE(client, SYSTEM_ERROR_NO_MEMORY);
+        CHECK(client->init(conf));
+        // Initialize network manager
+        mgr_.reset(new(std::nothrow) CellularNetworkManager(client.get()));
+        CHECK_TRUE(mgr_, SYSTEM_ERROR_NO_MEMORY);
+        client_ = std::move(client);
+        return 0;
+    }
+};
+
+} // unnamed
+
+WifiNetworkManager* wifiNetworkManager() {
+    static WifiNetworkManagerInit mgr;
+    return mgr.instance();
+}
+
+CellularNetworkManager* cellularNetworkManager() {
+    static CellularNetworkManagerInit mgr;
+    return mgr.instance();
+}
+
+} // particle
+
+int if_init_platform(void*) {
+    /* lo0 (created by LwIP) */
+
+    /* th1 - OpenThread (Deprecated) */
+    reserve_netif_index();
+
+    /* en2 - Ethernet FeatherWing (optional) */
+    uint8_t mac[6] = {0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff};
+    {
+        // const uint32_t lsb = __builtin_bswap32(NRF_FICR->DEVICEADDR[0]);
+        // const uint32_t msb = NRF_FICR->DEVICEADDR[1] & 0xffff;
+        // memcpy(mac + 2, &lsb, sizeof(lsb));
+        // mac[0] = msb >> 8;
+        // mac[1] = msb;
+        /* Drop 'multicast' bit */
+        mac[0] &= 0b11111110;
+        /* Set 'locally administered' bit */
+        mac[0] |= 0b10;
+    }
+
+    if (HAL_Feature_Get(FEATURE_ETHERNET_DETECTION)) {
+        en2 = new WizNetif(HAL_SPI_INTERFACE1, D5, D3, D4, mac);
+    }
+
+    uint8_t dummy;
+    if (!en2 || if_get_index(en2->interface(), &dummy)) {
+        /* No en2 present */
+        delete en2;
+        en2 = nullptr;
+        reserve_netif_index();
+    }
+
+#if !HAL_PLATFORM_WIFI_SCAN_ONLY
```

Comment:
This doesn't look right. You still need to have RealtekNcpNetif even if we are only allowing scanning.